app/Main.hs

```diff
@@ -23,61 +33,82 @@ data Command = MoveRight
              deriving (Show, Eq)
 
 tokenize :: String -> [Command]
-tokenize [] = []
-tokenize (x:xs)
-  | x == '>' = MoveRight : tokenize xs
-  | x == '<' = MoveLeft : tokenize xs
-  | x == '+' = IncCell: tokenize xs
-  | x == '-' = DecCell : tokenize xs
-  | x == '.' = PrintCell : tokenize xs
-  | x == ',' = InputCell: tokenize xs
-  | x == '[' = OpenBracket : tokenize xs
-  | x == ']' = CloseBracket: tokenize xs
-  | otherwise = tokenize xs
+tokenize = catMaybes . fmap charToCommand
+  where
+    charToCommand c =
+      case c of
+        '>' -> Just MoveRight
+        '<' -> Just MoveLeft
+        '+' -> Just IncCell
+        '-' -> Just DecCell
+        '.' -> Just PrintCell
+        ',' -> Just InputCell
+        '[' -> Just OpenBracket
+        ']' -> Just CloseBracket
+        _ -> Nothing
 
-
--- Sets a specific cell to a value
 setCell :: (Int -> Int) -> CellState -> Int -> CellState
-setCell f cs i = take i cs ++ [if j < 0 then 255 else j] ++ drop (i + 1) cs
-    where j = f $ cs !! i
+setCell _ [] _ = []
+setCell f (c:rest) 0 = f c : rest
+setCell f (c:rest) n = c : setCell f rest (n-1)
 
 incCell :: CellState -> Int -> CellState
 incCell = setCell (+1)
 
 decCell :: CellState -> Int -> CellState
-decCell = setCell (\x -> x - 1)
-
+decCell = setCell (subtract 1)
 
 -- Parse
 -- Close bracket is different from the others as it
 -- doesn't bind itself to the parse process
 parse :: [Command] -> StateT MoveState IO [Command]
-parse [] = StateT (\x -> pure ([], x))
-parse (x:xs) = if x == CloseBracket then s else s >>= parse
-    where s = StateT (\v@(i, pc, cs) ->
```

Comment:
You use `StateT` because you want to write programs in terms of `get` and `put`. For simple cases, you don't even need a monad transformer. 

In my PR I've rewritten this code using the `StateT` combinators, but if you think it is ugly and verbose then you should rewrite `parse` to have the type

```
parse :: [Commad] -> MoveState -> IO ([Command], MoveState)
```
And just completely cut out the transformer altogether.

It's fine because there are very few cases that you actually use `>>=`, in `do` notation or otherwise. So you can afford to carry around the State manually for such a small amount of code